OpenTelemetry (S3M) App Development apps (#1798)
SumoLogic/sumologic-documentation: docs/integrations/app-development/opentelemetry/jira.md

---
id: jira-opentelemetry
title: Jira - OpenTelemetry Collector
sidebar_label: Jira - OTel Collector
description: The Sumo Logic App for Jira provides insight into Jira user access, request activity, issues, security, sprint events, and user events.
---

import useBaseUrl from '@docusaurus/useBaseUrl';
import Tabs from '@theme/Tabs';
import TabItem from '@theme/TabItem';

> <img src={useBaseUrl('img/integrations/app-development/jira.png')} alt="Thumbnail icon" width="50"/> <img src={useBaseUrl('img/send-data/otel-color.svg')} alt="Thumbnail icon" width="45"/>

Reviewer: This image link is broken when I run it in a local build. Verify that it works in your system.

The Jira app uses the following log types:
-   **Jira Access Logs**. HTTP server access log files, in `logs/access_logs`,  in your JIRA Installation Directory.
-   **Jira Security Logs**.Security-related information, such as logins, logouts, session creation/destruction, and security denials from `atlassian-jira-security.log`.
-   **Jira Catalina Logs**. The application catalina log file, `logs/catalina.out`, in your JIRA Installation Directory.
-   **Jira Webhooks**. Webhook events of types:
    -   Issue
    -   User
    -   Sprint

## Configure hosted collector to receive Webhooks

In this step, you create a host collector to receive webhooks from Jira, and set up an HTTP source on it.

1.  Configure a [hosted collector](https://help.sumologic.com/docs/send-data/hosted-collectors/configure-hosted-collector/), or select an existing hosted collector for the HTTP source.
2.  Configure an [HTTP source](https://help.sumologic.com/docs/send-data/hosted-collectors/http-source/logs-metrics/) on the hosted collector.
    -   For Source Category, specify Atlassian/Jira/Events
    -   Add `sumo.datasource` as File to this source and assign the value **jira** for it.

>     <img src={useBaseUrl('https://sumologic-app-data-v2.s3.amazonaws.com/dashboards/Jira-OpenTelemetry/Jira-Datasource.png')} alt="Datasource" />

Reviewer: This image link is broken when I run it in a local build. Verify that it works in your system.

    -   Make a note of the HTTP address for the source. You will supply it when you configure a Jira webhook in the next step.

## Register webhook in Jira

Follow the instructions on [Webhooks](https://developer.atlassian.com/server/jira/platform/webhooks/) in Jira help to register a webhook for the following events:

-   Issue related:
    -   created (jira:issue_created)
    -   updated (jira:issue_updated)
    -   deleted (jira:issue_deleted)
    -   worklog changed (jira:worklog_updated)
-   User related:
    -   created (user_created)
    -   updated (user_updated)
    -   deleted (user_deleted)
-   Sprint related:
    -   created (sprint_created)
    -   deleted (sprint_deleted)
    -   updated (sprint_updated)
    -   started (sprint_started)
    -   closed (sprint_closed)

> When you configure the webhook, enter the URL for the HTTP source you created in [step 2](https://help.sumologic.com/docs/integrations/app-development/jira/#Step_2:_Configure_hosted_collector_to_receive_webhooks) as the endpoint for the webhook.

Reviewer: ```suggestion
When you configure the webhook, enter the URL for the [HTTP source you created](https://help.sumologic.com/docs/integrations/app-development/jira/#configure-hosted-collector-to-receive-webhooks) as the endpoint for the webhook.
```

Author: Hi @jpipkin1 , since we use relative links in docs, this would be `/docs/integrations/app-development/jira/#configure-hosted-collector-to-receive-webhooks`

## Logs Collection Configuration & App installation

As part of the setting up the collection process and app installation user can select the App from App Catalog and click on **Install App**. Please follow the steps below:

### Step 1: Set up Collector

If you want to create a new Collector please select **Add a new Collector** option.

:::note
If you want to use an existing OpenTelemetry Collector, you can skip this step by selecting the **Use an existing Collector** option.
:::

Select the platform for which you want to install the Sumo OpenTelemetry Collector.

This will generate a command which can be executed in the machine which needs to get monitored. Once executed it will install the Sumo Logic OpenTelemetry Collector agent.

> <img src={useBaseUrl('https://sumologic-app-data-v2.s3.amazonaws.com/dashboards/Jira-OpenTelemetry/Jira-Collector.png')} alt="Collector" />

Reviewer: This image link is broken when I run it in a local build. Verify that it works in your system.

### Step 2: Configure integration

In this step we will be configuring the yaml file required for the Jira Collection.

Path of the different log file configured to capture Jira logs is needed to be given here:

-   Jira Access Logs. Apache HTTP server log files. Default path `/var/log/apache2/*.log`.
-   Jira Security Logs. Security-related information. Default path `/home/jira/atlassian/application-data/jira/log/atlassian-jira-security.log`.
-   Jira Catalina Logs. The application server log file. Default path `/home/jira/atlassian/application-data/jira/log/*.log`.

Click on the **Download YAML File** button to get the yaml file.

> <img src={useBaseUrl('https://sumologic-app-data-v2.s3.amazonaws.com/dashboards/Jira-OpenTelemetry/Jira-YAML.png')} alt="YAML" />

Reviewer: This image link is broken when I run it in a local build. Verify that it works in your system.

### Step 3: Sending logs to Sumo

Once you have the yaml file downloaded in step 2, follow the below steps based on the platform of the machine :

Linux:

1.  Copy the yaml file to `/etc/otelcol-sumo/conf.d/` folder in the Jira instance which needs to be monitored.
2.  Place Env file in the following directory
```sh
/etc/otelcol-sumo/env/
```
3.  restart the collector using
```sh
 sudo systemctl restart otelcol-sumo
```

Windows:

1.  Copy the yaml file to `C:\ProgramData\Sumo Logic\OpenTelemetry Collector\config\conf.d` folder in the machine which needs to be monitored.
2.  Restart the collector using 
```sh
Restart-Service -Name OtelcolSumo
```

> Mac

Reviewer: ```suggestion
Mac:
```

Author: @jpipkin1 This section needs to be put in a table, as it is in the other companion docs.

---
id: jira-opentelemetry
title: Jira - OpenTelemetry Collector
sidebar_label: Jira - OTel Collector
description: The Sumo Logic App for Jira provides insight into Jira user access, request activity, issues, security, sprint events, and user events.
---

import useBaseUrl from '@docusaurus/useBaseUrl';
import Tabs from '@theme/Tabs';
import TabItem from '@theme/TabItem';

<img src={useBaseUrl('img/integrations/app-development/jira.png')} alt="Thumbnail icon" width="50"/> <img src={useBaseUrl('img/send-data/otel-color.svg')} alt="Thumbnail icon" width="45"/>

The Sumo Logic App for Jira provides insight into Jira usage, request activity, issues, security, sprint events, and user events. 

Jira logs are sent to Sumo Logic through OpenTelemetry [filelog receiver](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/receiver/filelogreceiver).

<img src={useBaseUrl('https://sumologic-app-data-v2.s3.amazonaws.com/dashboards/Jira-OpenTelemetry/Jira-Schematics.png')} alt="Schematics" />

## Fields creation in Sumo Logic for Jira

> Following are the Tags which will be created as part of Jira App install if not already present. 

Reviewer 2: ```suggestion
Following are the tags which will be created as part of the Jira App install if not already present. 
```

`sumo.datasource` - Has fixed value of **jira**.

## Prerequisites

This section provides instructions for configuring log collection for Jira running on a non-Kubernetes environment for the Sumo Logic App for Jira. Sumo Logic supports the collection of logs from Jira server in standalone environments.

Follow the instructions to set up log collection.

The Jira app uses the following log types:

> 

Reviewer 2: ```suggestion
```

-   **Jira Access Logs**. HTTP server access log files, in `logs/access_logs`,  in your JIRA Installation Directory.
-   **Jira Security Logs**.Security-related information, such as logins, logouts, session creation/destruction, and security denials from `atlassian-jira-security.log`.
-   **Jira Catalina Logs**. The application catalina log file, `logs/catalina.out`, in your JIRA Installation Directory.
-   **Jira Webhooks**. Webhook events of types:
    -   Issue
    -   User
    -   Sprint

## Configure hosted collector to receive Webhooks

In this step, you create a host collector to receive webhooks from Jira, and set up an HTTP source on it.

> 

Reviewer 2: ```suggestion
```

1.  Configure a [hosted collector](https://help.sumologic.com/docs/send-data/hosted-collectors/configure-hosted-collector/), or select an existing hosted collector for the HTTP source.
2.  Configure an [HTTP source](https://help.sumologic.com/docs/send-data/hosted-collectors/http-source/logs-metrics/) on the hosted collector.
    -   For Source Category, specify Atlassian/Jira/Events
    -   Add `sumo.datasource` as File to this source and assign the value **jira** for it.
    <img src={useBaseUrl('https://sumologic-app-data-v2.s3.amazonaws.com/dashboards/Jira-OpenTelemetry/Jira-Datasource.png')} alt="Datasource" />
    -   Make a note of the HTTP address for the source. You will supply it when you configure a Jira webhook in the next step.

## Register webhook in Jira

> 

Reviewer 2: ```suggestion
```

Follow the instructions on [Webhooks](https://developer.atlassian.com/server/jira/platform/webhooks/) in Jira help to register a webhook for the following events:

> 

Reviewer 2: ```suggestion
```

-   Issue related:
    -   created (jira:issue_created)
    -   updated (jira:issue_updated)
    -   deleted (jira:issue_deleted)
    -   worklog changed (jira:worklog_updated)
-   User related:
    -   created (user_created)
    -   updated (user_updated)
    -   deleted (user_deleted)
-   Sprint related:
    -   created (sprint_created)
    -   deleted (sprint_deleted)
    -   updated (sprint_updated)
    -   started (sprint_started)
    -   closed (sprint_closed)

When you configure the webhook, enter the URL for the HTTP source you created in [step 2](https://help.sumologic.com/docs/integrations/app-development/jira/#Step_2:_Configure_hosted_collector_to_receive_webhooks) as the endpoint for the webhook.

> ## Logs Collection Configuration & App installation

Reviewer 2: ```suggestion
## Logs Collection Configuration & App Installation
```

> As part of the setting up the collection process and app installation user can select the App from App Catalog and click on **Install App**. Please follow the steps below:

Reviewer 2: ```suggestion
As part of setting up the collection and app installation process, you can select the App from **App Catalog** and click on **Install App**.  Follow the steps below:
```

### Step 1: Set up Collector

> If you want to create a new Collector please select **Add a new Collector** option.

Reviewer 2: ```suggestion
 To create a new Collector, select **Add a new Collector** option.
```

Reviewer 2: Avoid using "please" in help docs.

:::note
If you want to use an existing OpenTelemetry Collector, you can skip this step by selecting the **Use an existing Collector** option.
:::

Select the platform for which you want to install the Sumo OpenTelemetry Collector.

This will generate a command which can be executed in the machine which needs to get monitored. Once executed it will install the Sumo Logic OpenTelemetry Collector agent.

<img src={useBaseUrl('https://sumologic-app-data-v2.s3.amazonaws.com/dashboards/Jira-OpenTelemetry/Jira-Collector.png')} alt="Collector" />

> ### Step 2: Configure integration

Reviewer 2: ```suggestion
### Step 2: Configure Integration
```

Reviewer 2: I'd recommend using lowercase letters in the heading only if there are fewer than four letters.

> In this step we will be configuring the yaml file required for the Jira Collection.

Reviewer 2: ```suggestion
In this step, we will be configuring the yaml file required for the Jira Collection.
```

Path of the different log file configured to capture Jira logs is needed to be given here:

> 

Reviewer 2: ```suggestion
```

-   Jira Access Logs. Apache HTTP server log files. Default path `/var/log/apache2/*.log`.
-   Jira Security Logs. Security-related information. Default path `/home/jira/atlassian/application-data/jira/log/atlassian-jira-security.log`.
-   Jira Catalina Logs. The application server log file. Default path `/home/jira/atlassian/application-data/jira/log/*.log`.

Click on the **Download YAML File** button to get the yaml file.

> <img src={useBaseUrl('https://sumologic-app-data-v2.s3.amazonaws.com/dashboards/Jira-OpenTelemetry/Jira-YAML.png')} alt="YAML" />

Reviewer 2: <br/><img src={useBaseUrl('https://sumologic-app-data-v2.s3.amazonaws.com/dashboards/Jira-OpenTelemetry/Jira-YAML.png')}

<br/> tag should be used when there is a screenshot just right after the step.

Reviewer: This image link is broken when I run it in a local build. Verify that it works in your system.

> ### Step 3: Sending logs to Sumo

Reviewer 2: ```suggestion
### Step 3: Sending logs to Sumo Logic
```
